amazon/ion/ion_py_objects.py

```diff
@@ -0,0 +1,616 @@
+from collections import OrderedDict
+from decimal import Decimal
+from collections.abc import MutableMapping
+
+from amazon.ion.core import IonType, IonEvent
+from amazon.ion.symbols import SymbolToken
+
+
+class IonPyNull_new(object):
+    __name__ = 'IonPyNull_new'
+    __qualname__ = 'IonPyNull_new'
+
+    def __init__(self, ion_type=IonType.NULL, value=None, annotations=()):
```

Comment:
Can we drop the `value` parameter? Or is it needed for consistency with the other `__init__` implementations?

Reply:
I tried to keep the constructor's signature same as `from_value`, so that anywhere which is using `from_value` can be in-place replaced with their constructor. But it's not necessarily, once the this PR is approved. I will create a separate PR targeting this one for clean up works.

Reply:
I tried to maintain the constructors' signatures identical to `from_value`'s, so that we can perform an in-place replacement. This approach can also help us quantify the overhead caused by `from_value`. For example, switching from `from_value` to their constructors introduces an 8-12% improvement in performance.

This change is not necessarily after the PR is merged. Once this PR is approved, I will create a separate PR targeting this one for all clean up works.